### PR TITLE
[WIP] Work In Progress - CSS refactoring

### LINKED ISSUE #410: CSS is currently centralized, should be componentized?
https://github.com/readium/readium-desktop/tree/develop/src/renderer/assets/styles

=> all in one folder, but should be tied to specific JSX component?

### PR DESCRIPTION
**Pull request by danielweck**
fixes #410 
Fixes https://github.com/edrlab/thorium-reader/issues/1196

### CODE REVIEW

**Comment by danielweck** [comment]
@ThibaultENT could you please fix the linting errors:

https://github.com/edrlab/thorium-reader/pull/1571/checks?check_run_id=3868774634

```
src/renderer/library/components/catalog/AboutThoriumButton.tsx
    58:28 ❌ unexpected trailing whitespace
                                            (EditorConfig trim_trailing_whitespace https://goo.gl/oo8iVb)
src/renderer/library/components/catalog/PublicationAddButton.tsx
    43:19 ❌ unexpected trailing whitespace
                                            (EditorConfig trim_trailing_whitespace https://goo.gl/oo8iVb)
    44:36 ❌ unexpected trailing whitespace
                                            (EditorConfig trim_trailing_whitespace https://goo.gl/oo8iVb)
src/renderer/library/components/opds/OpdsAddForm.tsx
    42:24 ❌ unexpected trailing whitespace
                                            (EditorConfig trim_trailing_whitespace https://goo.gl/oo8iVb)
src/renderer/library/components/settings/KeyboardSettings.tsx
    108:25 ❌ unexpected trailing whitespace       (EditorConfig trim_trailing_whitespace https://goo.gl/oo8iVb)
    111:01 ❌ invalid indent size: 21, expected: 20
                                                    (EditorConfig indent_size              https://goo.gl/QtGB8R)
    125:51 ❌ unexpected trailing whitespace       (EditorConfig trim_trailing_whitespace https://goo.gl/oo8iVb)
```

Thank you :)

**Comment by danielweck** [comment]
Hello @ThibaultENT I added comments above, do you see them in your GitHub notifications / emails? If not, can you please "subscribe" to this PR? (buttons on the right hand side)

File: src/renderer/library/components/catalog/TagLayout copy.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// ==LICENSE-BEGIN==
```
Comment (panaC) [inline src/renderer/library/components/catalog/TagLayout copy.tsx:1]: `TagLayout copy.tsx` is a wrong filename

File: src/renderer/library/components/opds/PageNavigation.tsx [hunk 2]
```diff
@@ -72,48 +75,62 @@ class PageNavigation extends React.Component<IProps, undefined> {
...
+                        to={{
+                            ...this.props.location,
+                            pathname: buildRoute(pageLinks.first[0]),
+                        }}
+                        className={styles.button_primary}
+                    >
+                        <SVG svg={ArrowFirstIcon} />
+                        {__("opds.firstPage")}
+                    </Link>
+                    :
```
Comment (danielweck) [inline src/renderer/library/components/opds/PageNavigation.tsx:88]: Incorrect syntax (compiler error)
Why a ternary expression?

File: src/renderer/library/components/opds/PageNavigation.tsx [hunk 3]
```diff
@@ -72,48 +75,62 @@ class PageNavigation extends React.Component<IProps, undefined> {
...
+                    <Link
+                        className={styles.button_primary}
+                        disabled={true}
```
Comment (danielweck) [inline src/renderer/library/components/opds/PageNavigation.tsx:91]: The `disabled` property is not valid on the `Link` JSX element

File: src/renderer/library/components/opds/PageNavigation.tsx (hunk 2 above)
Comment (danielweck) [reply src/renderer/library/components/opds/PageNavigation.tsx:88]: I fixed my comment above (typo). Please re-read :)

File: src/renderer/library/components/catalog/TagLayout copy.tsx (hunk 1 above)
Comment (danielweck) [reply src/renderer/library/components/catalog/TagLayout copy.tsx:1]: This file looks like a redundant / unused copy of the original `TagLayout.tsx`?

**Review by danielweck — COMMENTED** [review]
There are lots of CSS linting errors, I am shortly going to create a Pull Request containing all build fixes (including corrections to the inline comments in this PR)

**Comment by danielweck** [comment]
Proposed fixes: https://github.com/ThibaultENT/thorium-reader/pull/5

**Comment by danielweck** [comment]
Popup menu is clipped:
Screenshot 2021-11-04 at 10 54 34

**Comment by danielweck** [comment]
Breadcrumb is transparent, and obscures / overlays on top of scrolling content below:
Screenshot 2021-11-04 at 10 58 11

**Comment by danielweck** [comment]
Breadcrumb obscured:
Screenshot 2021-11-04 at 10 59 25

**Comment by danielweck** [comment]
Broken slide-in notifications:
Screenshot 2021-11-04 at 11 00 48

**Comment by danielweck** [comment]
Missing line break:
Screenshot 2021-11-04 at 11 03 04

**Comment by danielweck** [comment]
OPDS feed number should be right-aligned (or at least, separated from words):
Screenshot 2021-11-04 at 11 05 03

**Comment by danielweck** [comment]
OPDS browser regression: all navigation links are ABOVE the stream / grid of cover images (was on the left hand side before, if I remember correctly)
Screenshot 2021-11-04 at 11 08 36

**Comment by danielweck** [comment]
Blue keyboard focus outline on cover images is clipped:
Screenshot 2021-11-04 at 11 15 35

**Comment by danielweck** [comment]
Broken button layout:
Screenshot 2021-11-04 at 11 20 08
Screenshot 2021-11-04 at 11 17 46

**Comment by danielweck** [comment]
In "publication info" modal popup dialog, publications without cover images destroy the component layout:
Screenshot 2021-11-04 at 11 19 05

**Comment by danielweck** [comment]
Broken button layout in reader window:
Screenshot 2021-11-04 at 11 32 42

**Comment by danielweck** [comment]
In reader window, navigation menu now displays rows / stacked items with less spacing:
Screenshot 2021-11-04 at 11 35 40
Screenshot 2021-11-04 at 11 35 15

File: src/renderer/library/components/opds/Header.tsx [hunk 4]
```diff
@@ -196,7 +202,8 @@ class Header extends React.Component<IProps, undefined> {
                         ...this.props.location,
                         pathname: route,
                     }}
-                    style={{ marginLeft: "16px" }}
+                    className={classNames(styles.button_transparency_icon, styles.button_refresh)}
+                    style={ {transform: "scale(.85)"} }
```
[the remark this answers is not in the document]
Comment (danielweck) [reply src/renderer/library/components/opds/Header.tsx]: Closing old comment.

File: src/renderer/library/components/catalog/TagLayout copy.tsx (hunk 1 above)
Comment (danielweck) [reply src/renderer/library/components/catalog/TagLayout copy.tsx]: fixed

File: src/renderer/library/components/opds/PageNavigation.tsx (hunk 2 above)
Comment (danielweck) [reply src/renderer/library/components/opds/PageNavigation.tsx]: fixed

File: src/renderer/library/components/opds/PageNavigation.tsx (hunk 3 above)
Comment (danielweck) [reply src/renderer/library/components/opds/PageNavigation.tsx]: closing outdated code comment

File: src/renderer/common/components/dialog/publicationInfos/formatPublisherDate.tsx [hunk 5]
```diff
@@ -26,9 +26,9 @@ export const FormatPublisherDate: React.FC<IProps> = (props) => {
         const date = moment(publication.publishedAt).format("L");
         if (date) {
             formatedPublishedDateComponent = (
-                <p>
+                <div>
```
Comment (panaC) [inline src/renderer/common/components/dialog/publicationInfos/formatPublisherDate.tsx:29]: I'm not an expert, but I've heard that a div alone is misunderstood by screen reader.
@danielweck what do you think ?

Comment (panaC) [reply src/renderer/common/components/dialog/publicationInfos/formatPublisherDate.tsx:29]: It's just for my curiosity :) .. not a strict review comment

File: src/renderer/library/components/opds/Entry copy.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
+// ==LICENSE-BEGIN==
```
Comment (panaC) [inline src/renderer/library/components/opds/Entry copy.tsx:1]: `Entry copy.tsx` but why ? :(

**Comment by panaC** [comment]
```
git pull css develop
git checkout css/develop
git clean
npm install
npm run start:dev
```

```

ERROR in src/renderer/common/components/dialog/publicationInfos/FormatContributorWithLink.tsx:39:56
TS2322: Type 'string' is not assignable to type 'number'.
    37 |                 retElement.push(
    38 |                     <a onClick={onClickLinkCb(newContributor)}
  > 39 |                         className={styles.button_link} tabIndex="0"
       |                                                        ^^^^^^^^
    40 |                     >
    41 |                         {translator.translateContentField(newContributor.name)}
    42 |                     </a>,

ERROR in src/renderer/common/components/dialog/publicationInfos/PublicationInfoDescription.tsx:82:86
TS2339: Property 'description' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    80 |                             <div
    81 |                                 ref={this.descriptionRef}
  > 82 |                                 className={classNames(styles.allowUserSelect, styles.description)}
       |                                                                                      ^^^^^^^^^^^
    83 |                                 dangerouslySetInnerHTML={{__html: textSanitize}}
    84 |                             >
    85 |                             </div>

ERROR in src/renderer/common/components/dialog/publicationInfos/tag/AddTag.tsx:48:32
TS2339: Property 'flux_search' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    46 |                 ? <form
    47 |                     onSubmit={this.addTag}
  > 48 |                     id={styles.flux_search}
       |                                ^^^^^^^^^^^
    49 |                 >
    50 |                     <input
    51 |                         type="text"

ERROR in src/renderer/library/components/layout/LibraryLayout.tsx:85:15
TS6133: '__' is declared but its value is never read.
    83 |     public render() {
    84 |         const { title } = this.props;
  > 85 |         const { __ } = this.props;
       |               ^^^^^^
    86 |
    87 |         let helmetTitle = capitalizedAppName;
    88 |         if (title) {

ERROR in src/renderer/library/components/opds/Browser.tsx:40:58
TS2339: Property 'opdsBrowseBreadcrumb' does not exist on type '{ readonly publicationgrid: string; readonly entrygroups: string; readonly opds_form: string; readonly opds_list: string; readonly opds_entry_ul: string; readonly flux_infos: string; ... 9 more ...; readonly page_count: string; }'.
    38 |     public render(): React.ReactElement<IProps>  {
    39 |         const secondaryHeader = <Header/>;
  > 40 |         const breadCrumb = <BreadCrumb className={styles.opdsBrowseBreadcrumb} breadcrumb={this.props.breadrumb} />;
       |                                                          ^^^^^^^^^^^^^^^^^^^^
    41 |
    42 |         return (
    43 |             <LibraryLayout

ERROR in src/renderer/library/components/opds/BrowserResult.tsx:164:71
TS2339: Property 'flux_slider' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    162 |                                                 ) === DisplayType.Grid ?
    163 |                                                 <Slider
  > 164 |                                                     className={styles.flux_slider}
        |                                                                       ^^^^^^^^^^^
    165 |                                                     content={group.publications.map((pub, pubId) =>
    166 |                                                         <PublicationCard
    167 |                                                             key={`opds-group-${i}-${pubId}`}

ERROR in src/renderer/library/components/opds/BrowserResult.tsx:202:39
TS2339: Property 'opdsBrowseContent' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    200 |         }
    201 |
  > 202 |         return <div className={styles.opdsBrowseContent}>
        |                                       ^^^^^^^^^^^^^^^^^
    203 |             {content}
    204 |         </div>;
    205 |     }

ERROR in src/renderer/library/components/opds/Header.tsx:128:83
TS2339: Property 'button_small' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    126 |                         pathname: route,
    127 |                     }}
  > 128 |                     className={classNames(styles.button_transparency_icon, styles.button_small)}
        |                                                                                   ^^^^^^^^^^^^
    129 |                 >
    130 |                     <SVG svg={AvatarIcon} title={__("opds.shelf")} />
    131 |                 </Link>

ERROR in src/renderer/library/components/opds/Header.tsx:165:83
TS2339: Property 'button_small' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    163 |                         pathname: route,
    164 |                     }}
  > 165 |                     className={classNames(styles.button_transparency_icon, styles.button_small)}
        |                                                                                   ^^^^^^^^^^^^
    166 |                 >
    167 |                     <SVG svg={HomeIcon} title={__("header.homeTitle")} />
    168 |                 </Link>

ERROR in src/renderer/library/components/opds/Header.tsx:204:106
TS2339: Property 'button_small' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    202 |                         pathname: route,
    203 |                     }}
  > 204 |                     className={classNames(styles.button_transparency_icon, styles.button_refresh, styles.button_small)}
        |                                                                                                          ^^^^^^^^^^^^
    205 |                 >
    206 |                     <SVG svg={RefreshIcon} title={__("header.refreshTitle")} />
    207 |                 </Link>

ERROR in src/renderer/library/components/opds/Header.tsx:215:106
TS2339: Property 'button_small' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    213 |                         ...this.props.location,
    214 |                     }}
  > 215 |                     className={classNames(styles.button_transparency_icon, styles.button_refresh, styles.button_small)}
        |                                                                                                          ^^^^^^^^^^^^
    216 |                 >
    217 |                     <SVG svg={RefreshIcon} title={__("header.refreshTitle")} />
    218 |                 </Link>

ERROR in src/renderer/library/components/opds/PageNavigation.tsx:91:33
TS2322: Type '{ children: (string | Element)[]; className: string; tabindex: string; }' is not assignable to type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'.
  Property 'tabindex' does not exist on type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'. Did you mean 'tabIndex'?
    89 |                             <Link
    90 |                                 className={classNames(styles.button_primary, styles.disabled)}
  > 91 |                                 tabindex="-1"
       |                                 ^^^^^^^^
    92 |                             >
    93 |                                 <SVG svg={ArrowFirstIcon} />
    94 |                                 {__("opds.firstPage")}

ERROR in src/renderer/library/components/opds/PageNavigation.tsx:112:33
TS2322: Type '{ children: (string | Element)[]; className: string; tabindex: string; }' is not assignable to type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'.
  Property 'tabindex' does not exist on type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'. Did you mean 'tabIndex'?
    110 |                             <Link
    111 |                                 className={classNames(styles.button_primary, styles.disabled)}
  > 112 |                                 tabindex="-1"
        |                                 ^^^^^^^^
    113 |                             >
    114 |                                 <SVG svg={ArrowLeftIcon} />
    115 |                                 {__("opds.previous")}

ERROR in src/renderer/library/components/opds/PageNavigation.tsx:123:48
TS2339: Property 'page_count' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    121 |                     && pageInfo.numberOfItems
    122 |                     && pageInfo.itemsPerPage
  > 123 |                     && <span className={styles.page_count}>
        |                                                ^^^^^^^^^^
    124 |                         {
    125 |                             pageInfo.currentPage
    126 |                         } / {

ERROR in src/renderer/library/components/opds/PageNavigation.tsx:147:33
    205 |                 >
    206 |                     <SVG svg={RefreshIcon} title={__("header.refreshTitle")} />
    207 |                 </Link>

ERROR in src/renderer/library/components/opds/Header.tsx:215:106
TS2339: Property 'button_small' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    213 |                         ...this.props.location,
    214 |                     }}
  > 215 |                     className={classNames(styles.button_transparency_icon, styles.button_refresh, styles.button_small)}
        |                                                                                                          ^^^^^^^^^^^^
    216 |                 >
    217 |                     <SVG svg={RefreshIcon} title={__("header.refreshTitle")} />
    218 |                 </Link>

ERROR in src/renderer/library/components/opds/PageNavigation.tsx:91:33
TS2322: Type '{ children: (string | Element)[]; className: string; tabindex: string; }' is not assignable to type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'.
  Property 'tabindex' does not exist on type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'. Did you mean 'tabIndex'?
    89 |                             <Link
    90 |                                 className={classNames(styles.button_primary, styles.disabled)}
  > 91 |                                 tabindex="-1"
       |                                 ^^^^^^^^
    92 |                             >
    93 |                                 <SVG svg={ArrowFirstIcon} />
    94 |                                 {__("opds.firstPage")}

ERROR in src/renderer/library/components/opds/PageNavigation.tsx:112:33
TS2322: Type '{ children: (string | Element)[]; className: string; tabindex: string; }' is not assignable to type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'.
  Property 'tabindex' does not exist on type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'. Did you mean 'tabIndex'?
    110 |                             <Link
    111 |                                 className={classNames(styles.button_primary, styles.disabled)}
  > 112 |                                 tabindex="-1"
        |                                 ^^^^^^^^
    113 |                             >
    114 |                                 <SVG svg={ArrowLeftIcon} />
    115 |                                 {__("opds.previous")}

ERROR in src/renderer/library/components/opds/PageNavigation.tsx:123:48
TS2339: Property 'page_count' does not exist on type '{ readonly main: string; readonly breadcrumb: string; readonly primary: string; readonly anchor_link: string; readonly allowUserSelect: string; readonly rotate180: string; readonly fw_bold: string; ... 131 more ...; readonly toRemove: string; }'.
    121 |                     && pageInfo.numberOfItems
    122 |                     && pageInfo.itemsPerPage
  > 123 |                     && <span className={styles.page_count}>
        |                                                ^^^^^^^^^^
    124 |                         {
    125 |                             pageInfo.currentPage
    126 |                         } / {

ERROR in src/renderer/library/components/opds/PageNavigation.tsx:147:33
TS2322: Type '{ children: (string | Element)[]; className: string; tabindex: string; }' is not assignable to type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'.
  Property 'tabindex' does not exist on type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'. Did you mean 'tabIndex'?
    145 |                             <Link
    146 |                                 className={classNames(styles.button_primary, styles.disabled)}
  > 147 |                                 tabindex="-1"
        |                                 ^^^^^^^^
    148 |                             >
    149 |                                 {__("opds.next")}
    150 |                                 <SVG svg={ArrowRightIcon} />

ERROR in src/renderer/library/components/opds/PageNavigation.tsx:168:33
TS2322: Type '{ children: (string | Element)[]; className: string; tabindex: string; }' is not assignable to type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'.
  Property 'tabindex' does not exist on type 'IntrinsicAttributes & LinkProps<unknown> & RefAttributes<HTMLAnchorElement>'. Did you mean 'tabIndex'?
    166 |                             <Link
    167 |                                 className={classNames(styles.button_primary, styles.disabled)}
  > 168 |                                 tabindex="-1"
        |                                 ^^^^^^^^
    169 |                             >
    170 |                                 {__("opds.lastPage")}
    171 |                                 <SVG svg={ArrowLastIcon} />
```

**Comment by panaC** [comment]
> In reader window, navigation menu now displays rows / stacked items with less spacing:
> Screenshot 2021-11-04 at 11 35 40
> Screenshot 2021-11-04 at 11 35 15


same issue
image

**Comment by panaC** [comment]
image

night mode in develop branch : 

image

**Comment by panaC** [comment]
image



Blue keyboard focus on thorium : 
image

**Comment by panaC** [comment]
image

in thorium : 

image
image

File: src/renderer/common/components/dialog/publicationInfos/formatPublisherDate.tsx (hunk 5 above)
Comment (danielweck) [reply src/renderer/common/components/dialog/publicationInfos/formatPublisherDate.tsx:29]: No, it is fine to use `div` as non-semantic, purely structural dividers. :)

File: src/renderer/library/components/opds/Entry copy.tsx (hunk 6 above)
Comment (danielweck) [reply src/renderer/library/components/opds/Entry copy.tsx:1]: Indeed, can be removed?

**Comment by danielweck** [comment]
As expected, the clickable author/publisher and blue tags works fine, but only in OPDS2 Feedbooks (not OPDS1):

Screenshot 2021-11-05 at 16 49 40
Screenshot 2021-11-05 at 16 50 17


...however I note that the blue tags are clickable in OPDS1, even though they are not actual links (they are “dead”)
Is it possible to remove the “proactive feedback” (mouse hover blue effect and hand mouse cursor) when the tags are not hyperlinks?
(like the author/publishers, etc.)

**Comment by danielweck** [comment]
Minor visual "bug" / inconsistency (blue keyboard focus outline):
Screenshot 2021-11-05 at 16 44 05
Screenshot 2021-11-05 at 16 43 32

**Comment by danielweck** [comment]
Minor issue with clipped / shrunk keyboard outline:

Screenshot 2021-11-05 at 16 59 45

**Comment by ThibaultENT** [comment]
> As expected, the clickable author/publisher and blue tags works fine, but only in OPDS2 Feedbooks (not OPDS1):
> 
> Screenshot 2021-11-05 at 16 49 40
> 
> Screenshot 2021-11-05 at 16 50 17
> 
> ...however I note that the blue tags are clickable in OPDS1, even though they are not actual links (they are “dead”) Is it possible to remove the “proactive feedback” (mouse hover blue effect and hand mouse cursor) when the tags are not hyperlinks? (like the author/publishers, etc.)
Capture d’écran 2021-11-06 à 11 45 40

Ces conditions sont là depuis le début.
→ Dois-je les retirer ?
→ Dois-je remplacer les <span... par des <a onClick... ?

Je rappelle qu'avant non-plus il n'existait pas de lien sur les deux 'else', mais seulement sur la première condition.

**Comment by danielweck** [comment]
PR superseded by https://github.com/edrlab/thorium-reader/pull/1589
(NPM/Node upgrade, requires package updates)

**Pull request closed without merging**